src/legacy/ui/public/vis/editors/default/agg_param.js

```diff
@@ -105,10 +96,13 @@ uiModules
                 $scope.showValidation = true;
               }
             }, true);
+
             $scope.paramValue = $scope.agg.params[$scope.aggParam.name];
           }
 
           $scope.onChange = (value) => {
+            $scope.paramValue = value;
+
             // This is obviously not a good code quality, but without using scope binding (which we can't see above)
             // to bind function values, this is right now the best temporary fix, until all of this will be gone.
             $scope.$parent.onParamChange($scope.agg, $scope.aggParam.name, value);
```

Comment:
I see you are assigning $scope.paramValue, but if I am following this control flow correctly, that appears redundant. Is this necessary?

Reply:
You're right, is't redundant.
Fixed.